src/service/cid-impl.js

```diff
@@ -170,12 +173,17 @@ export class Cid {
    * @return {!Promise<?string>}
    */
   getExternalCid_(getCidStruct, persistenceConsent) {
+    const scope = getCidStruct.scope;
     /** @const {!Location} */
     const url = parseUrl(this.ampdoc.win.location.href);
     if (!isProxyOrigin(url)) {
+      const apiClient =
+          ViewerCidApi.scopeOptedInForCidApi(this.ampdoc.win, scope);
+      if (apiClient) {
+        return this.cidApi_.getScopedCid(scope, apiClient);
```

Comment:
isn't that wrong? If the clientId is stored in AMP_ECID_GOOGLE then analytics.js will have to rerequest the client id from the API. I would expect the clientId to be written to _ga cookie.